### PR TITLE
Makes you able to butt out cigarettes on people

### PR DESCRIPTION
**Pull request by Geatish**
## About The Pull Request

-You can now butt cigarettes on people by being on harm intent. 
-Attacking someone on non harm intent will reduce its time lit and has a chance to be put out. 

## Why It's Good For The Game

Adding more actions to cigarettes and cigars good.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/130587823/55a837f0-2b73-4c93-9793-fef4dc3492f2

</details>

## Changelog
:cl: Geatish
tweak: you can now butt cigarettes on other people if you're on harm intent
tweak: cigarettes will have their smoke time reduced and have a chance to get extinguished if attacking on non harm intent
/:cl:

### CODE REVIEW

**Comment by ColonelOrion** [comment]
Why don't you add spitting on people while you're at it.
Fucking insufferable.

This will be a boon to cunty shits and encourage general ic assholery.

**Comment by Geatish** [comment]
Chill dude 😭, this wasn't directed at you. i'm just trying to learn how to code and i wanted to do something trivial. 
I don't think it really is a big deal, yea some asshole might burn you with their cigarette, but that's armed combat already so just crit his ass.

**Comment by ColonelOrion** [comment]
@Geatish sorry man, yeah, that was too much.

**Comment by Fronsis** [comment]
> Chill dude 😭, this wasn't directed at you. i'm just trying to learn how to code and i wanted to do something trivial. I don't think it really is a big deal, yea some asshole might burn you with their cigarette, but that's armed combat already so just crit his ass.

I mean, critting someone because they burnt you with a ciggie might be too much it's like critting someone just for a single jab on the chin.

Well, regardless this shouldn't do much damage IMO, i'd say the same burn level damage as when touching a lightbulb, i agree with Orion though but welp, more reasons to arrest people for Assault i guess.

**Comment by tontyGH** [comment]
> This may encourage general IC assholery

Yes. That is roleplay.

File: code/game/objects/items/cigs_lighters.dm [hunk 1]
```diff
@@ -297,6 +297,21 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		qdel(src)
 	. = ..()
 
+/obj/item/clothing/mask/cigarette/afterattack(mob/M, mob/user, proximity)
```
Comment (tontyGH) [inline code/game/objects/items/cigs_lighters.dm]: There is already an afterattack() defined for cigarettes around line 180

Although double definitions like this would compile and work, it's considered bad practice and it makes code flow somewhat hard to follow

**Review by tontyGH — CHANGES_REQUESTED** [review]

**Comment by MarkusLarsson421** [comment]
> On further thought a lot of people might find this OOC uncomfortable depending on the context.

Not the blood, gore, cannabalism, poisoning, explosives, armed combat, and such? Cigarette butting...

**Comment by ColonelOrion** [comment]
> Yes. That is roleplay.
Ah yes "It's what my character would do!"
The standard paen of people looking for excuses to be cunty in an rp setting.

**Comment by MarkusLarsson421** [comment]
> Ah yes "It's what my character would do!"
> The standard paen of people looking for excuses to be cunty in an rp setting.

There are plenty of options of being "cunty" with or without this, you're overreacting so calm down. SS13 is no utopia. You know that just as much as I do.

**Comment by ColonelOrion** [comment]
Letting people act a certain way with mechanics encourages behaviour.

Why would we want to encourage the kind of shitty characters that would even use this mechanic?

**Comment by MarkusLarsson421** [comment]
You're overreacting.

**Comment by tontyGH** [comment]
> Ah yes "It's what my character would do!"
> 
> The standard paen of people looking for excuses to be cunty in an rp setting.

Yes. This is a roleplaying server. People will want to roleplay as bad people. Not everybody on-board the station is a goody two-shoes.

This isn't like "Well, my character was paranoid so he would steal HoS gun" or "my character is evil so he would harmbaton a man for no reason", but a simple roleplaying action that minimally harms the recipient and adds more character expression. It's **flavour text**.

You're getting heated over BM. It's genuinely not that bad. Chill out.

**Comment by Geatish** [comment]
> Well, regardless this shouldn't do much damage IMO, i'd say the same burn level damage as when touching a lightbulb-
When you're butting it out on someone I'm only making it dubble, which is just 8 burn damage. A normal punch does 7 damage for reference. So like this won't ever have any combat advantage whatsoever.

**Comment by MarkusLarsson421** [comment]
> A normal punch does 7 damage for reference.

I don't know about you, but I don't think it should do equal or more damage than a punch.

If its mainly for RP, then even 1 to 2 damage is enough.

**Comment by Geatish** [comment]
> I don't know about you, but I don't think it should do equal or more damage than a punch.

I mean, i think it would hurt more to get a cigarette put out on you than to be punched. Plus I strongly doubt that someone will be lighting up a new cigarette each time they want to hurt someone in a fight.

**Comment by Fronsis** [comment]
> > A normal punch does 7 damage for reference.
> 
> I don't know about you, but I don't think it should do equal or more damage than a punch.
> 
> If its mainly for RP, then even 1 to 2 damage is enough.

That's why i suggested the burn damage should be trivial like the  ''touching a bulb'' damage (i think it's super low, can't remember the number)  since it's just a flavor thing to be ''rude ic''.

What hurts more IRL doesn't matter since there's different factors to take into account for that scenario.

**Comment by Geatish** [comment]
> That's why i suggested the burn damage should be trivial like the ''touching a bulb'' damage

The damage is trivial, using a whole cigarette on someone for 8 burn damage isn't worth it in terms of combat. Also when attacking someone on non harm intent it will still do its standard 4 damage, plus there will be 40% chance that the cigarette will just extinguish and you'll have to relight it.

File: code/game/objects/items/cigs_lighters.dm [hunk 2]
```diff
@@ -297,19 +297,33 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 		qdel(src)
 	. = ..()
 
-/obj/item/clothing/mask/cigarette/afterattack(mob/M, mob/user, proximity)
+/obj/item/clothing/mask/cigarette/afterattack(mob/living/M, mob/living/user, proximity)
 	if(!istype(M))
 		return ..()
+	if(HAS_TRAIT(user, TRAIT_PACIFISM))
```
Comment (Gilgaxx) [inline code/game/objects/items/cigs_lighters.dm]: I cant test it right now, but is this check for the pacifism trait needed? It should already stop the attack chain during the attack() proc right?

Comment (Geatish) [reply code/game/objects/items/cigs_lighters.dm]: Since I added extra damage for butting it out on someone it would bypass the pacifist trait and add the 4 extra damage anyway, it would also trigger the code to tell it to remove smoke time and extinguish the cigarette which i thought didn't make sense if you're not hurting anyone.

**Review by Rukofamicom — APPROVED** [review]
Seems like a good interaction to add.

**Comment by Rukofamicom** [comment]
To address the major criticism here:

> This may encourage general ic assholery.

Yes it definitely will. **IC** assholery should never be seen as a problem - it's OOC assholery that is a problem. 

> Why would we want to encourage the kind of shitty characters that would even use this mechanic?

Because character conflict is what creates and drives interesting rounds, especially ahead of antagonists getting involved. A rude and imperfect character is a compelling one that creates minor conflicts for others to build on and create an interesting story/dynamic in the round. 

It most notably provides a hook for antagonistic players to lean into and incorporate the help (direct or indirect) of upset, but normal crew in several cases.

**Comment by PestoVerde322** [comment]
I chime in and say that i'm pretty much in line with what Ruko above said. And after all, what's just a bit of banter that Security can easly break up and the med team patch up? 
If the the assilant or the defender perish due to the rampant escalation, bwoink the winner about "overescalation". This feels very much like an IC solvable case in case of IC assholery. 
Other than that, i'd personally knock down the damage from 4 burn to 2 burn, simply due to then being easier to patch out by eating some food.

The only case i can think this is bad (with the current damage) if someone get quite literally Ceasar'd by 8 other people by constantly litting up a cigarette, extinguish them on the target, and repeat untill you run out of cigarettes (6 cigarettes per packs, multiplied by 4 and then again multiplied by 8 is a whopping 192 damage!)

**Comment by Geatish** [comment]
> The only case i can think this is bad (with the current damage) if someone get quite literally Ceasar'd by 8 other people by constantly litting up a cigarette, extinguish them on the target, and repeat untill you run out of cigarettes (6 cigarettes per packs, multiplied by 4 and then again multiplied by 8 is a whopping 192 damage!)

this is just a nothing burger, a punch does 7 damage, one less than butting out a cigarette on someone. IF you were to light up a cigarette and but it on them each time then youd need over 2 packs of cigarettes, plus the extra amount of time you need to take one out and light it up, its just faster to punch someone a extra two times instead of having to light up cigarettes all the time. 
Also, if someone would take the time and effort to fill their backpack with lit cigarettes to have a slight advantage is just ????. I really don't see cigarette butt chaining becoming some new combat meta.

File: code/game/objects/items/cigs_lighters.dm (hunk 1 above)
Comment (Geatish) [reply code/game/objects/items/cigs_lighters.dm]: made each def a proc and added a new def sorting it out.

**Comment by DarnTheMarn** [comment]
I like this, I tried to have a Reservoir Dogs type moment a while back where I had someone tied to a chair and covered them in flammable booze and tried to put a cigarette out on them to ignite it like a cool super villain. Only for it not to work whether I threw it, poked it, jabbed it, you name it. Having to use the lighter was pretty funny but also ruined the moment a little.

**Review by Rukofamicom — COMMENTED** [review]
Code looks good now as well, and this has gotten positive reception from admins when presented as potentially creating behavior problems.
